src/client/network.rs

```diff
@@ -371,6 +387,7 @@ impl KafkaConnection {
                id: id,
                host: host.to_owned(),
                stream: stream,
+               closed: false,
```

Comment:
it just a mark for the connection pool when we found the server closed the connection. If you like to trace the whole lifecycle of connection, I could add a new state machine for it.